### DISCUSSION TITLE
Improvements to Renovating Internal Dependencies

### DISCUSSION THREAD

**Discussion by zackerydev** [renovatebot/renovate#10373] [Suggest an Idea]
Hi All 👋 

Recently a feature was added to also renovate internal Lerna/Yarn/Npm Workspaces dependencies (`updateInternalDeps`)

This feature works great and allows us to not worry about a significant portion of our manual effort when managing dependencies. However there could still be an improvement to save some time when used with `"bumpVersion": "patch"`

Right now the flow works like this:
- RenovateBot comes in to renovate an internal package's outdated `lodash` dependency. It raises a PR and bumps `@internal/utils` package to `1.0.1`.
- This PR is merged and then a separate PR is opened by RenovateBot where the consumers of `@internal/utils` are bumped to `1.0.1`

The improvement to this flow would be for RenovateBot to handle both in the same PR. Example here: 
https://github.com/zgriesinger/lerna-monorepo/pull/5/files#diff-10c66822af6659cfc94a96d933dc56740119076f4c7ffee78fd4d2b5464a3819L28

The primary reason for this is in the workspaces environment you move from remote installs to symlinks in the back and forth with the current flow. This would keep things symlinked at all times and reduce package-lock thrashing.

Let me know thoughts on this!

**Comment by rarkins** [comment]
Aren't you missing a middle step in "the flow", which is "Someone publishes `@internal/utils@1.0.1`"? Otherwise if `1.0.1` is not on your registry then I don't see how the last step would happen.

**Reply by viceice** [reply]
I think this is a follow up from  #6887

**Reply by rarkins** [reply]
Yes, my question is assuming that. Renovate gets its list of versions from registries, not from internal package.json files

**Reply by zackerydev** [reply]
We do have the publish step for outside consumers, however for packages that consume other packages internally to the mono repo a package can consume the new version _before_ its published.

Lerna looks at the versions and if they match they get symlinked locally instead of pulling from remote.

So bumping the package to `1.0.1` _and_ the consumer to consuming `1.0.1` can work (and works quite nicely) before the `1.0.1` version exists.

Its primarily useful when doing changes for an app and a package simultaneously. A good example would be when upgrading react. For example, if you have an app on React 16 and it consumes a package internal to the mono repo on React 16, RenovateBot would currently upgrade the whole mono repo to React 17 but that would result in the app pulling in React 17 and the previous version of the internal package pulling in React 16 until that next Renovate PR was created renovating the internal dependencies.

I can make a more concrete example repo, but that's the thought process.
